docs/requirements.rst

```diff
@@ -0,0 +1,115 @@
+Requirements
+------------
+
+Hard requirements
+^^^^^^^^^^^^^^^^^
+
+DeepChem currently supports Python 3.5 through 3.7 and requires these packages on any condition.
+
+- `joblib`_
+- `NumPy`_
+- `pandas`_
+- `scikit-learn`_
+- `SciPy`_
+- `TensorFlow`_
+
+  - `deepchem>=2.4.0` requires tensorflow v2
+  - `deepchem<2.4.0` requires tensorflow v1
+
+
+Soft requirements
```

Comment:
I think this is not better. This list should be managed automatically as much as possible.
Does anyone have a idea...?

Reply:
This looks beautiful! 

It would be nice to automatically manage this list, but I'm not entirely sure how to. The Model cheatsheet is another similarly large table that currently has to maintained by hand

Reply:
Can you link to this from `docs/index.rst` so we get this rendered on the docs table of contents?